### PR TITLE
Add Confluence advanced sync rules

### LINKED ISSUE #954: add support for advanced sync rules as defined in the document (also referred to as remote filtering)

### PR DESCRIPTION
**Pull request by praveen-elastic**
## Closes #954 

This PR introduces advanced sync rules for Confluence by using CQL (Confluence Query Language) queries to filter out confluence data. The connector utilises `atlassian.py` file for validating the query.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Comment by seanstory** [comment]
Should this also add sync rule docs for confluence here https://github.com/elastic/connectors-python/tree/main/docs/sync-rules ?

File: connectors/sources/confluence.py [hunk 1]
```diff
@@ -442,6 +449,58 @@ async def fetch_attachments(
...
+                        document.update(
+                            {
+                                "size": entity_details.get("extensions", {}).get(
+                                    "fileSize"
+                                ),
+                                container: entity_details.get("container", {}).get(
+                                    "title"
+                                ),
+                            }
+                        )
```
Comment (timgrein) [inline connectors/sources/confluence.py:498]:
```suggestion
                    file_size =  entity_details.get("extensions", {}).get("fileSize")
                    title = entity_details.get("container", {}).get("title")
                    document.update(
                         {
                             "size": file_size,
                             "container": title
                         }
                     )
```

I think using variables would improve readability here. Also container should be in quotes, right?

File: connectors/sources/confluence.py [hunk 2]
```diff
@@ -442,6 +449,58 @@ async def fetch_attachments(
...
+                        document.pop("body")
```
Comment (timgrein) [inline connectors/sources/confluence.py:499]:
```suggestion
                        # Removing body as attachment will be downloaded lazily
                        document.pop("body")
```

nit: Maybe a comment would be useful, why we're removing the body

File: connectors/sources/confluence.py [hunk 3]
```diff
@@ -557,25 +616,42 @@ async def get_docs(self, filtering=None):
...
+
         else:
-            quoted_spaces = "','".join(self.spaces)
-            configured_spaces_query = f"cql=space in ('{quoted_spaces}') AND type="
-        await self.fetchers.put(self._space_coro)
-        await self.fetchers.put(
-            partial(
-                self._page_blog_coro,
-                f"{configured_spaces_query}blogpost&{CONTENT_QUERY}",
+            if self.spaces == ["*"]:
```
Comment (timgrein) [inline connectors/sources/confluence.py:635]:
```suggestion
            if self.spaces == [WILDCARD]:
```

suggestion: I think we should use a constant here as `*` could mean anything

File: connectors/sources/confluence.py (hunk 1 above)
Comment (praveen-elastic) [reply connectors/sources/confluence.py]: Agreed. I've updated with variable names to improve readability. Also for container variable I've renamed it to `container_type` as it denotes the type of container.

File: connectors/sources/confluence.py (hunk 2 above)
Comment (praveen-elastic) [reply connectors/sources/confluence.py:498]: Good catch, updated [cc07dee](https://github.com/elastic/connectors-python/pull/956/commits/cc07deeb07c1485d5df46659357d0be281cc2a1c)

**Review by timgrein — APPROVED** [review]
LGTM!

**Pull request merged**